### DISCUSSION TITLE
Where to see the Profile name and ID mapping

### DISCUSSION THREAD

**Discussion by xkeshav** [microsoft/vscode-discussions#1927] [Extension Development QnA] [answered]
Ubuntu 24.04
VS Code 1.94.1

New[ Profile Editor](https://code.visualstudio.com/docs/editor/profiles) is very good feature but it creates a little bit confusion; I can name and add icons to the the profile but when I open the individual profile's _settings.json_ ,notice that the path of these _settings.json_ prefixed with some `~/.config/Code/profiles-<id>/`  

so I want to know where can we see the profile id anme name mapping within vscode or some other way?

for example:

I have 3 Custom Profile name aalong with default 

- Desktop
- Ubuntu
- Mac

and when I open Profile Editor and open the settings of any profile it shows the settings.json from path

`/home/<username>/.config/Code/User/profiles` and there are folder name such as 

- -4b2326d3
- -38eb64c9
- -59ee9f2d
- -579e772c

![image](https://github.com/user-attachments/assets/4024bd09-459d-418c-81cd-00aafa48dc77)

My issue is I am not able to find which profile belong to which, one way is open settings file and see the path name but is there any any location where this mapping is written?

Thanks in Advance

**Comment by alefragnani** [comment, marked as answer]
Hi @xkeshav ,

There is no way to _map_ between the profile `name` and its `id`, as the `id` is generated from a hash of a UUID. Also, there is no API available, yet, to expose the Profiles.

I have already opened a feature request for an API to access Profiles (https://github.com/microsoft/vscode/issues/211890). Feel free to upvote and/or add comments if to improve the request, or open a new one with your use case.

Hope this helps
